src/Raven.Server/Documents/Indexes/Index.cs

```diff
@@ -508,9 +508,9 @@ internal void HandleWriteErrors(IndexingStatsScope stats, IndexWriteException iw
             if (iwe.InnerException is SystemException) // Don't count transient errors
                 return;
 
-            writeErrors = Interlocked.Increment(ref writeErrors);
+            _writeErrors = Interlocked.Increment(ref _writeErrors);
 
-            if (Priority.HasFlag(IndexingPriority.Error) || Interlocked.Read(ref writeErrors) < WriteErrorsLimit)
+            if (Priority.HasFlag(IndexingPriority.Error) || Interlocked.Read(ref _writeErrors) < WriteErrorsLimit)
```

Comment:
Here should use the _local variable_ holding the current value of the write errors